pkg/plugin/restore_pvc_action_plugin.go

```diff
@@ -94,6 +94,41 @@ func (p *NewPVCRestoreItemAction) Execute(input *velero.RestoreItemActionExecute
 			UpdatedItem: item,
 		}, nil
 	}
+
+	ctx := context.Background()
+	veleroNs, exist := os.LookupEnv("VELERO_NAMESPACE")
+	if !exist {
+		errMsg := "Failed to lookup the ENV variable for velero namespace"
+		p.Log.Error(errMsg)
+		return nil, errors.New(errMsg)
+	}
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		p.Log.Errorf("Failed to get the rest config in k8s cluster: %v", err)
+		return nil, errors.WithStack(err)
+	}
+
+	backupdriverClient, err := backupdriverTypedV1.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	// Since the volume snapshotter plugin is also registered, the restore path of volume snapshotter is triggered.
+	// At this point of code, we are certain that we are restoring 1.1.0 backup since there is annotation present.
+	// Verifying that vsl-vsphere does not exist will ensure restore path is not triggered twice.
+	vsphereVolumeSnapLocationName, err := utils.CheckForVolumeSnapshotLocation(ctx, restConfig, veleroNs, p.Log)
+	if err != nil {
+		p.Log.Errorf("Failed to check VolumeSnapshotLocation for vsphere provider.")
+		return nil, errors.WithStack(err)
+	}
+	if vsphereVolumeSnapLocationName != "" {
+		errMsg := "Found VolumeSnapshotLocations " + vsphereVolumeSnapLocationName + " for vsphere provider, please delete it and try again.."
+		p.Log.Errorf(errMsg)
+		p.Log.Warnf("NOTE: Deleting VolumeSnapshotLocations will disable the plugins ability to restore snapshots from plugin versions lesser than 1.1.0")
+		return nil, errors.New(errMsg)
```

Comment:
Please evaluate whether it is better to return `SkipRestore: true` as below.

Reply:
removed this code.